staging/vhost-device-gpu/src/main.rs

```diff
@@ -0,0 +1,140 @@
+// VIRTIO GPU Emulation via vhost-user
+//
+// Copyright 2024 Red Hat Inc
+//
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use log::{error, info};
+use std::{path::PathBuf, process::exit};
+
+use clap::{Parser, ValueEnum};
+use thiserror::Error as ThisError;
+use vhost_device_gpu::{
+    device::{self, VhostUserGpuBackend},
+    GpuConfig, GpuMode,
+};
+use vhost_user_backend::VhostUserDaemon;
+use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, ThisError)]
+pub(crate) enum Error {
+    #[error("Could not create backend: {0}")]
+    CouldNotCreateBackend(device::Error),
+    #[error("Could not create daemon: {0}")]
+    CouldNotCreateDaemon(vhost_user_backend::Error),
+    #[error("Fatal error: {0}")]
+    ServeFailed(vhost_user_backend::Error),
+}
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct GpuArgs {
+    /// vhost-user Unix domain socket.
+    #[clap(short, long, value_name = "SOCKET")]
+    socket_path: PathBuf,
+    #[clap(short, long, value_enum)]
+    renderer: RenderMode,
+}
+
+#[derive(Debug, Clone, ValueEnum)]
+enum RenderMode {
```

Comment:
This is a duplicate of what you have in lib.rs, any reason for not re-using that one?